src/Node.h

```diff
@@ -103,7 +103,11 @@ class WIN32_EXPORT Node {
             return nullptr;
         }
 
-        ::connect(fd, result->ai_addr, result->ai_addrlen);
+        if (::connect(fd, result->ai_addr, result->ai_addrlen) == -1) {
```

Comment:
Yeah, probably best to just write if (connect) since that is less platform specific